utils/generate-commands-json.py

```diff
@@ -0,0 +1,120 @@
+    rep = {}
+    for f in flags:
+        rep[f] = True
+    return rep
+
+def convert_argument(arg):
+    rep = convert_list_to_dict(arg, False)
+    if 'flags' in rep:
+        rep = {
+            **rep,
+            **convert_flags_to_truthy_dict(rep['flags'])
+        }
+        del rep['flags']
+    if 'arguments' in rep:
+        rep['arguments'] = list(map(lambda x: convert_argument(x), rep['arguments']))
+    return rep
+
+def convert_keyspec(spec):
+    rep = convert_list_to_dict(spec, False)
+    if 'flags' in rep:
+        rep = {
+            **rep,
+            **convert_flags_to_truthy_dict(rep['flags'])
+        }
+        del rep['flags']
+    rep['begin_search'] = convert_list_to_dict(rep['begin_search'], True)
+    rep['find_keys'] = convert_list_to_dict(rep['find_keys'], True)
+    return rep
+
+def convert_command_to_objects(container, cmd):
+    obj = {}
+    rep = [obj]
+    name = cmd[0].upper()
+    key = f'{container} {name}' if container else name
+
+    meta = convert_list_to_dict(cmd[7], False)
+    meta['arity'] = cmd[1]
+    meta['command_flags'] = cmd[2]
+    meta['acl_categories'] = cmd[6]
+
+    if 'key_specs' in meta:
+        meta['key_specs'] = list(map(lambda x: convert_keyspec(x), meta['key_specs']))
+    if 'arguments' in meta:
+        meta['arguments'] = list(map(lambda x: convert_argument(x), meta['arguments']))
+    if 'subcommands' in meta:
+        sub = list(map(lambda x: convert_command_to_objects(name, x), meta['subcommands']))
+        rep += [s for sc in sub for s in sc]
+    if 'doc_flags' in meta:
+        meta['doc_flags'] = convert_flags_to_truthy_dict(meta['doc_flags'])
+    else:
+        meta['doc_flags'] = {}
+
+    FIELDS = [
+        'summary',
+        'since',
+        'group',
+        'complexity',
+        'deprecated_since',
+        'replaced_by',
+        'history',
+        'acl_categories',
+        'arity',
+        'key_specs',
+        'arguments',
+        'command_flags',
+    ]
+    obj[key] = {}
+    for field in FIELDS:
+        if field in meta and (type(meta[field]) is not list or len(meta[field]) > 0):
+            obj[key][field] = meta[field]
+
+    obj[key] = {
+        **obj[key],
+        **meta['doc_flags'],
+    }
+    return rep
+
+# MAIN
+
+# Parse arguments
+parser = argparse.ArgumentParser(description="generate commands.json from a Redis server")
+parser.add_argument("-u", "--uri", type=str, help="the server's URI", default="redis://default:@localhost:6379")
+args = parser.parse_args()
+
+# Get `COMMAND`'s output
+r = redis.Redis().from_url(args.uri, decode_responses=True)
+command_output = r.command()
+r.close()
+
+# Transform the output
+commands = list(map(lambda x: convert_command_to_objects(None,x), command_output))
+commands = [command for cmd in commands for command in cmd]
+commands.sort(key=lambda x: list(x.keys())[0])
+
+payload = {}
+for cmd in commands:
+    name = list(cmd.keys())[0]
+    payload[name] = cmd[name]
+
+# Output JSON
```

Comment:
Maybe that json needs to contain the redis version is extracted the info from? 
We can later maybe rename the file manually or the caller script, but sine were already connected to the source, maybe we should extract some info from it and store it in the json as Metadata?

Reply:
So something like:

```
{
  "version": "7.0.0",
  "timestamp": 12387657486,
  "commands": {
    "ACL": { ... },
    ...
  }
}
```

Reply:
maybe include these fields from INFO SERVER:
```
            "redis_version:%s\r\n"
            "redis_git_sha1:%s\r\n"
            "redis_git_dirty:%i\r\n"
            "redis_build_id:%s\r\n"
```
add timestamp, and put them inside some "version" node?

Reply:
why do we need a timestamp? the git sha is our "timestamp"

Reply:
just since we're recording how it was generated, so we might as well also record when and by who.

but i'm actually not sure if all this metadata belongs in the file or not.
maybe it'll just be confusing and it should be recorded in some way in the file name (i.e. commands.redis7.json)